### ISSUE TITLE
proposal: prefer any instead of for loop

### ISSUE THREAD

**Issue by bernaferrari** [dart-lang/linter#3358]
## Description

```dart
bool isTwo = false;
for (final String child in [1, 2, 3, 4]) {
  if (child.toString() == "2") {
    isTwo = true;
    break;
  }
}
// OR
bool isTwo = false;
[1, 2, 3, 4].forEach((child) {
  if (child.toString() == "2") {
    isTwo = true;
  }
}
```
becomes
```dart
bool isTwo = [1, 2, 3, 4].any((child) => child.toString() == "2");
```
## Details

I believe using `any` is simpler and clearer than a for loop + boolean variable (sometimes even more efficient), but people might not know it.

## Kind

This only helps.

**Comment by srawlins** [comment]
I don't think we can reliably report which loops can be converted to use Iterable.any. The loop could contain multiple statements. It could contain multiple if statements with `var found = true; break;`, which could be deeply nested.

This rule is probably too complex to warrant the effort and maintenance.
